statements.tex

```diff
@@ -11,79 +11,96 @@
 %
-and for a range-based for statement of the form
+\oldoldtxt{and for a range-based for statement of the form}
 
 \begin{bnf}
-\terminal{for} \terminal{(} for-range-declaration \terminal{:} braced-init-list \terminal{)} statement
+\oldoldtxt{\terminal{for} \terminal{(} for-range-declaration \terminal{:} braced-init-list \terminal{)} statement}
 \end{bnf}
 %
-let \grammarterm{range-init} be equivalent to the \grammarterm{braced-init-list}.
+\oldoldtxt{let \grammarterm{range-init} be equivalent to the \grammarterm{braced-init-list}.}
+\end{removedblock}
 %
-In each case, a range-based for statement is equivalent to
+\oldoldtxt{In each case, a range-based for statement} is equivalent to
+
 \begin{removedblock}
-\begin{codeblock}
-{
-  auto && __range = range-init;
-  for ( auto __begin = begin-expr,
-             __end = end-expr;
-        __begin != __end;
-        ++__begin ) {
-    @\grammarterm{for-range-declaration}@ = *__begin;
-    @\grammarterm{statement}@
-  }
-}
-\end{codeblock}
+\begin{ncbnftab}
+\terminal{\{}\br
+\>\terminal{auto \&\&__range =} range-init \terminal{;}\br
+\>\terminal{for ( auto __begin =} begin-expr \terminal{,}\br
+\>\>\terminal{ __end =} end-expr \terminal{;}\br
+\>\>\terminal{ __begin != __end; ++__begin ) \{}\br
+\>\>for-range-declaration \terminal{= *__begin;}\br
+\>\>statement\br
+\>\terminal{\}}\br
+\terminal{\}}
+\end{ncbnftab}
 \end{removedblock}
+
 \begin{addedblock}
-\begin{codeblock}
-{
-  auto && __range = @\grammarterm{range-init}@;
-  auto __begin = @\grammarterm{begin-expr}@;
-  auto __end = @\grammarterm{end-expr}@;
-  for ( ; __begin != __end; ++__begin ) {
-    @\grammarterm{for-range-declaration}@ = *__begin;
-    @\grammarterm{statement}@
-  }
-}
-\end{codeblock}
+\begin{ncbnftab}
+\terminal{\{}\br
+\>\terminal{auto \&\&__range =} \oldoldtxt{range-init}\newnewtxt{for-range-initializer} \terminal{;}\br
+\>\terminal{auto __begin =} begin-expr \terminal{;}\br
+\>\terminal{auto __end =} end-expr \terminal{;}\br
+\>\terminal{for ( ; __begin != __end; ++__begin ) \{}\br
+\>\>for-range-declaration \terminal{= *__begin;}\br
+\>\>statement\br
+\>\terminal{\}}\br
+\terminal{\}}
+\end{ncbnftab}
 \end{addedblock}
-%
-where \tcode{__range}, \tcode{__begin}, and \tcode{__end} are variables defined for
-exposition only, and \tcode{_RangeT} is the type of the
-\grammarterm{}{expression}, and \textit{begin-expr} and \textit{end-expr} are
-determined as follows:
+
+where
+
+\begin{itemize}
+\item
+\newnewtxt{if the \grammarterm{for-range-initializer} is an \grammarterm{expression},
+it is regarded as if it were surrounded by parentheses (so that a comma operator
+cannot be reinterpreted as delimiting two \grammarterm{init-declarator}{s});}
+
+\item
+\tcode{__range}, \tcode{__begin}, and \tcode{__end} are variables defined for
+exposition only, and \oldoldtxt{\tcode{_RangeT} is the type of the
+\grammarterm{}{expression}, and}
+
+\item
+\placeholder{begin-expr} and \placeholder{end-expr} are determined as follows:
```

Comment:
My thought was that the line break and bullet are purely editorial differences, and that I should minimize the markup to make the overall result easier to review. That said, someone will be comparing the text to 14 and putting the old/new markup in will prevent them wasting a minute or so figure out what's going on here. Will change.